### ISSUE TITLE
[feature-request] Special case E721 (use isinstance instead of type comparison) for `type(None)`

### ISSUE THREAD

**Issue by smackesey** [charliermarsh/ruff#1056]
Comparison to `type(None)` occasionally comes up in our codebase:

```
if context.dagster_type.typing_type == type(None):
    return None
```

This raises a ruff/flake8 error: `E721 Do not compare types, use isinstance()`.

IMO this rule should not apply for comparisons to `type(None)`, since the "correct" form implied by the rule looks very odd:

```
if isinstance(None, context.dagster_type.typing_type):
    return None
```

I think this weirdness is unique to `type(None)`.
